src/ui/public/agg_types/buckets/geo_hash.js

```diff
@@ -114,29 +108,29 @@ export const geoHashBucketAgg = new BucketAggType({
       controller: function () {
       },
       write: function (aggConfig, output) {
-        const vis = aggConfig.vis;
-        const currZoom = getMapZoom(vis);
+        const currZoom = aggConfig.params.mapZoom;
         const autoPrecisionVal = zoomPrecision[currZoom];
-        output.params.precision = aggConfig.params.autoPrecision ? autoPrecisionVal : getPrecision(aggConfig.params.precision);
+        if (autoPrecisionVal) {
+          output.params.precision = aggConfig.params.autoPrecision ?
+            autoPrecisionVal : getPrecision(aggConfig.params.precision);
+        }
       }
     }
   ],
   getRequestAggs: function (agg) {
     const aggs = [];
+    const { vis, params } = agg;
 
-    if (agg.params.isFilteredByCollar && agg.getField()) {
-      const vis = agg.vis;
-      const mapBounds = vis.sessionState.mapBounds;
```

Comment:
What was the motivation to move this out of `vis.sessionState`?

Reply:
`vis.sessionState` is completely removed as it won't work with our future plans